scrapy_google_sheets_exporter/gsheets_exporter.py

```diff
@@ -0,0 +1,79 @@
+import csv
+import logging
+from io import StringIO
+
+import gspread
+from scrapy.exceptions import NotConfigured
+from scrapy.extensions.feedexport import BlockingFeedStorage, build_storage
+
+logger = logging.getLogger(__name__)
+
+
+class GoogleSheetsFeedStorage(BlockingFeedStorage):
+    def __init__(self, uri, credentials, *, feed_options=None):
+        if credentials:
+            self.gc = gspread.service_account_from_dict(credentials)
+        else:
+            raise NotConfigured(
+                "Must specify GOOGLE_CREDENTIALS (dict) in the spider settings."
+            )
+
+        self.spreadsheet_key, self.sheet_name = self.parse_gsheets_uri(uri)
+        self.spreadsheet = self.gc.open_by_key(self.spreadsheet_key)
+        self.sheet = self.spreadsheet.worksheet(self.sheet_name)
+        self.feed_options = feed_options or {}
+        self.append_mode = self.feed_options.get("append_mode", False)
+        self.fields = self.feed_options.get("fields", [])
+        self.format = self.feed_options.get("format", "csv")
```

Comment:
Format should always be CSV or CSV-compatible when using this FeedStorage. Otherwise your code below that requires loading the file as CSV won't work.

Reply:
Yes I'm aware @VMRuiz, that's why I added a `NotConfigured` error right below that line of code to warn the user if the format provided is not csv.. I also added as an instruction the repo's README, do you suggest I add anything else? thank you!

Reply:
Didn't noticed that, great!